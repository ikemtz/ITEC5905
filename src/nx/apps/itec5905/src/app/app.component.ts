import { Component } from '@angular/core';
import { OidcFacade, OidcUserFacade, IOidcUser } from 'imng-oidc-client';
import { Observable } from 'rxjs';

@Component({
  selector: 'itec-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
})
export class AppComponent {
  public readonly identity$: Observable<IOidcUser | undefined>;
  public readonly email$: Observable<string | undefined>;
  public readonly loggedIn$: Observable<boolean>;
  public readonly profilePicture$: Observable<string | undefined>;
  constructor(
    public readonly oidcFacade: OidcFacade,
    private readonly oidcUserFacade: OidcUserFacade
  ) {
    this.identity$ = this.oidcFacade.identity$;
    this.email$ = this.oidcUserFacade.email$;
    this.loggedIn$ = this.oidcFacade.loggedIn$;
    this.profilePicture$ = this.oidcUserFacade.profilePicture$;
  }

  public login(): void {
    this.oidcFacade.signinRedirect();
  }
}
