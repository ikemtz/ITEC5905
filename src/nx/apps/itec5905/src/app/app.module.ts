import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { RouterModule } from '@angular/router';
import { AppComponent } from './app.component';
import { appRoutes } from './app.routes'; 
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import { StoreRouterConnectingModule } from '@ngrx/router-store';
import { ImngOidcClientModule } from 'imng-oidc-client';
import { environment } from '../environments/environment';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { HomeComponent } from './home/home.component';

@NgModule({
  declarations: [AppComponent, HomeComponent],
  imports: [
    BrowserModule,
    RouterModule.forRoot(appRoutes, { initialNavigation: 'enabledBlocking' , enableTracing: true}),
    StoreModule.forRoot(
      {},
      {
        metaReducers: [],
        runtimeChecks: {
          strictActionSerializability: environment.runtimeChecks.strictActionSerializability,
          strictActionTypeUniqueness: environment.runtimeChecks.strictActionTypeUniqueness,
          strictStateSerializability: environment.runtimeChecks.strictStateSerializability,
          strictActionImmutability: environment.runtimeChecks.strictActionImmutability,
          strictStateImmutability: environment.runtimeChecks.strictStateImmutability,
        },
      }
    ),
    EffectsModule.forRoot([]),    
    StoreDevtoolsModule.instrument({ name: 'ITEC5905' }),
    StoreRouterConnectingModule.forRoot(),
    ImngOidcClientModule.forRoot(environment.oidc_options),
  ],
  providers: [],
  bootstrap: [AppComponent],
})
export class AppModule {}
