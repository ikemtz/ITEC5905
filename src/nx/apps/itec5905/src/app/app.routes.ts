import { Route } from '@angular/router';
import {  oidcRoutes } from 'imng-oidc-client';
import { HomeComponent } from './home/home.component';

export const appRoutes: Route[] = [ {
    path: 'artists',
    loadChildren: () =>
      import('./modules/artists-module/artists.module').then(
        (m) => m.ArtistsModule,
      ),
    //canActivateChild: [AuthGuard],
  },
  ...oidcRoutes,
  { path: '**', component: HomeComponent },];
