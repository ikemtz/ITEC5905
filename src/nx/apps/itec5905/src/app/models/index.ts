export * from './album-ext';
export * from './album-upsert';
export * from './artist-ext';
export * from './artist-upsert';
export * from './song-ext';
export * from './song-upsert';
