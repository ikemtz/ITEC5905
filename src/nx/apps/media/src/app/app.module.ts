import { Module } from '@nestjs/common';
import { MediaController } from './media.controller';
import { PinataService } from './pinata.service';
import { HttpModule } from '@nestjs/axios';
import { ImageController } from './image.controller';

@Module({
  imports: [HttpModule],
  controllers: [MediaController, ImageController],
  providers: [PinataService],
})
export class AppModule { }
