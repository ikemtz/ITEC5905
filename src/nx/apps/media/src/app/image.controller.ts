import { Controller, Get, Logger, Query } from '@nestjs/common';
import { ApiOkResponse, ApiTags, ApiQuery } from '@nestjs/swagger';
import { HttpService } from '@nestjs/axios';
import { Observable, map, tap } from 'rxjs';

@ApiTags('Images')
@Controller('image')
export class ImageController {
  constructor(private readonly httpService: HttpService) { }

  @Get()
  @ApiOkResponse({ type: String })
  @ApiQuery({ name: 'fileType', type: String })
  @ApiQuery({ name: 'ipfsHash', type: String })
  public getFile(@Query() query: { ipfsHash: string, fileType: string }): Observable<{ data: string }> {
    return this.httpService.get<string>(`https://gateway.pinata.cloud/ipfs/${query.ipfsHash}`, {
      headers: {
        pinata_api_key: process.env.pinataApiKey,
        pinata_secret_api_key: process.env.pinataSecretApiKey,
        Accept: "text/plain",
      }
    })
      .pipe(
        //tap(x => Logger.log(x.data)),
        map(x => ({ data: x.data })));
  }
}
