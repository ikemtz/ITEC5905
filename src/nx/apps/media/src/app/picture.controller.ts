import {
  Controller,
  Get,
  Logger,
  Query,
  Res,
  HttpStatus,
  StreamableFile,
} from '@nestjs/common';
import { ApiOkResponse, ApiTags, ApiQuery } from '@nestjs/swagger';
import { HttpService } from '@nestjs/axios';
import { Observable, map, tap } from 'rxjs';

import { Response } from 'express';

@ApiTags('Pictures')
@Controller('picture')
export class PictureController {
  private readonly regex = /^data:([a-z\/]*)/y;
  constructor(private readonly httpService: HttpService) { }

  @Get()
  @ApiOkResponse({ type: String })
  @ApiQuery({ name: 'fileType', type: String })
  @ApiQuery({ name: 'ipfsHash', type: String })
  public getFile(
    @Query() query: { ipfsHash: string; fileType: string },
    @Res() res: Response
  ): Observable<StreamableFile> {
    const fileType = this.regex.exec(query.fileType)[1];
    return this.httpService
      .get<string>(`https://gateway.pinata.cloud/ipfs/${query.ipfsHash}`, {
        headers: {
          pinata_api_key: process.env.pinataApiKey,
          pinata_secret_api_key: process.env.pinataSecretApiKey,
          Accept: 'text/plain',
        },
      })
      .pipe(
        map((x) => {
          const binaryString = atob(x.data);
          const bytes = new Uint8Array(binaryString.length);
          for (let i = 0; i < binaryString.length; i++) {
            bytes[i] = binaryString.charCodeAt(i);
          }
          return bytes;
        }),

        map(
          (x) =>
            new StreamableFile(x, { type: fileType })
        )
      );
  }
}
