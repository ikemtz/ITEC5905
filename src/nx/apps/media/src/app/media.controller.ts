import { Body, Controller, Get, Post, Query } from '@nestjs/common';
import { PinataService } from './pinata.service';
import { ApiOkResponse, ApiOperation, ApiTags, ApiQuery } from '@nestjs/swagger';
import { MediaUploadRequest } from '../models/media-upload-request';
import { MediaDataResponse } from '../models/media-meta-data';
import { UploadFileResponse } from '../models/upload-file-response';

@ApiTags('Media')
@Controller('media')
export class MediaController {
  constructor(private readonly pinataService: PinataService) { }

  @Get()
  @ApiOkResponse({ type: MediaDataResponse, isArray: true })
  @ApiQuery({ name: 'ipfsHash', type: String })
  public async getFiles(@Query('ipfsHash') ipfsHash: string | undefined): Promise<MediaDataResponse[]> {
    const result = await this.pinataService.getFiles(ipfsHash);
    return result.map(x => (
      {
        ipfsHash: x.ipfs_pin_hash,
        fileSize: x.size,
        fileName: x.metadata.name.toString(),
        createdOnUtc: new Date(x.date_pinned),
        referenceId: x.metadata.refId.toString(),
        referenceName: x.metadata.refName.toString(),
      }
    ))
  }

  @Post()
  @ApiOperation({
    description: 'Uploads a file to IPFS and returns an IPFS hash (CID) of the uploaded file',
  })
  @ApiOkResponse({ type: UploadFileResponse })
  public async uploadFile(@Body() request: MediaUploadRequest): Promise<UploadFileResponse> {
    const result = await this.pinataService.uploadFile(request);
    return {
      ipfsHash: result.IpfsHash,
      fileSize: result.PinSize,
      createdOnUtc: new Date(result.Timestamp)
    };
  }
}
