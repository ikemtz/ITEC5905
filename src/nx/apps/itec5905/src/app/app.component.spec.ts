import { TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import { RouterTestingModule } from '@angular/router/testing';
import { of } from 'rxjs';
import { OidcFacade, OidcUserFacade } from 'imng-oidc-client';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      declarations: [AppComponent],
      providers: [
        { provide: OidcFacade, useValue: { loggedIn$: of(true) } },
        { provide: OidcUserFacade, useValue: { email$: of('test@email.com'), profilePicture$: of('test.png') } }]
    }).compileComponents();
  });

  it(`Should match snapshot`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
