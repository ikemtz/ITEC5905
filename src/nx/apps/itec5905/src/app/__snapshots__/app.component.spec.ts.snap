// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent Should match snapshot 1`] = `
<div
  id="root0"
>
  <nav
    class="navbar navbar-expand-lg bg-secondary"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="#"
      >
        ITEC 5905
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          class="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            class="nav-item"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="#"
            >
              Home
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
            >
              Artists
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              role="button"
            >
               Dropdown 
            </a>
            <ul
              class="dropdown-menu"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Action
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Another action
                </a>
              </li>
              <li>
                <hr
                  class="dropdown-divider"
                />
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="#"
                >
                  Something else here
                </a>
              </li>
            </ul>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link disabled"
            >
              Disabled
            </a>
          </li>
        </ul>
        <form
          class="d-flex"
          role="search"
        >
          
          
        </form>
      </div>
    </div>
  </nav>
  <router-outlet />
  
</div>
`;
